util/cloud/s3.h

```diff
@@ -25,7 +28,10 @@ class S3Bucket {
   //! Called with (size, key_name) pairs.
   using ListObjectCb = std::function<void(size_t, std::string_view)>;
 
-  std::error_code ListObjects(std::string_view path, ListObjectCb cb);
+  ListObjectsResult ListObjects(std::string_view path, ListObjectCb cb,
+                                std::string_view marker = "", int max_keys = 1000);
+
+  std::error_code ListAllObjects(std::string_view path, ListObjectCb cb);
```

Comment:
add a comment saying that this a convenience function that handles pagination internally and iterates over all matching items.